test/jdk/java/util/zip/ZipFile/InvalidBytesInEntryNameOrComment.java

```diff
@@ -21,163 +21,304 @@
  * questions.
  *
  */
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
 
-import org.testng.annotations.BeforeTest;
-import org.testng.annotations.Test;
-
-import java.io.ByteArrayOutputStream;
+import java.io.FileInputStream;
 import java.io.FileOutputStream;
 import java.io.IOException;
 import java.nio.ByteBuffer;
 import java.nio.ByteOrder;
-import java.nio.channels.FileChannel;
+import java.nio.file.Files;
 import java.nio.file.Path;
 import java.util.Arrays;
-import java.util.zip.ZipEntry;
-import java.util.zip.ZipException;
-import java.util.zip.ZipFile;
-import java.util.zip.ZipOutputStream;
+import java.util.Formatter;
+import java.util.stream.Stream;
+import java.util.zip.*;
 
-import static org.testng.Assert.assertEquals;
-import static org.testng.Assert.expectThrows;
+import static org.junit.jupiter.api.Assertions.*;
 
 /**
  * @test
- * @summary Validate that opening ZIP files files with invalid UTF-8
- * byte sequences in the name or comment fields fails with ZipException
- * @run testng/othervm InvalidBytesInEntryNameOrComment
+ * @bug 8301873 8321156
+ * @summary Validate that a ZipException is thrown when opening a ZIP file via
+ * ZipFile, traversing a Zip File via ZipInputStream, with invalid UTF-8
+ * byte sequences in the name or comment fields fails with ZipException.
```

Comment:
The leading summary sentence needs a cleanup. Here's one suggestion:

```suggestion
 * @summary Validate that a ZipException is thrown when a ZIP file with 
 * invalid UTF-8 byte sequences in the name or comment fields is opened via
 * ZipFile or traversed via ZipInputStream.
```

Reply:
Updated